src/components/marketplace/cancel_order_button.tsx

```diff
@@ -0,0 +1,68 @@
+import { SignedOrder } from '@0x/connect';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import { cancelOrder } from '../../store/actions';
+import { UIOrder } from '../../util/types';
+
+interface OwnProps {
+    order: UIOrder;
+}
+
+interface DispatchProps {
+    onCancelOrder: (order: SignedOrder) => Promise<any>;
+}
+
+type Props = OwnProps & DispatchProps;
+
+interface State {
+    isLoading: boolean;
+}
+
+const BUTTON = styled.button`
+    background: none;
+    border: none;
+    color: #C4C4C4;
+    display: inline-block;
+    height: 1em;
+    outline: 0;
+
+    &:hover {
+        cursor: pointer;
+    }
+`;
+
+class CancelOrderButton extends React.Component<Props, State> {
+    public state = {
+        isLoading: false,
+    };
+
+    public render = () => {
+        const { isLoading } = this.state;
+        return <BUTTON type="button" disabled={isLoading} onClick={this._cancelOrder}>
+            <FontAwesomeIcon icon="times" />
+        </BUTTON>;
+    };
+
+    private readonly _cancelOrder  = async () => {
```

Comment:
I'm not against this, but it would be nice to have a convention here. Should we use `private readonly` and an underscore prefix for all methods that are not lifecycle hooks? Pros: it makes sense. Cons: more keywords.

I'm inclined to say yes. And if we do it, maybe we could add a linter rule for that (if there is one).

Reply:
In this case, I saw 2 linting errors:
* `Private member variable '_cancelOrder' is never reassigned; mark it as 'readonly'.`
* `private and protected members must be prefixed with an underscore`

Reply:
Cool, so we only have to "remember" to make them private, since I found no rule that enforces making those methods private.